src/Core/Context/LegacyControllerContext.php

```diff
@@ -0,0 +1,189 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Context;
+
+use Media;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Traversable;
+
+/**
+ * This class ensures compatibility with the context controller in pages migrated to Symfony.
+ * It encompasses the majority of public fields found in a legacy controller.
+ */
+class LegacyControllerContext
+{
+    public ?string $className;
+
+    /**
+     * List of CSS files.
+     *
+     * @var string[]
+     */
+    public array $css_files = [];
+
+    /**
+     * List of JavaScript files.
+     *
+     * @var string[]
+     */
+    public array $js_files = [];
+
+    // Controller type. Possible values: 'front', 'modulefront', 'admin', 'moduleadmin'.
+    public string $controller_type;
+
+    // Controller name.
+    public string $php_self;
+
+    /**
+     * Errors displayed after post processing
+     *
+     * @var array<string|int, string|bool>
+     */
+    public array $errors = [];
+
+    public array $warnings = [];
+
+    public array $informations = [];
+
+    public array $confirmations = [];
+
+    public string $currentIndex;
+
+    public int $id = -1;
+
+    // Security token
+    public ?string $token;
+
+    public string $override_folder;
+
+    // Image type
+    public string $imageType = 'jpg';
+
+    // Current controller name without suffix
+    public string $controller_name;
+
+    public int $multishop_context = -1;
+
+    // Bootstrap variable
+    public array|Traversable $page_header_toolbar_btn = [];
+
+    public array $_conf;
+
+    public array $_error;
```

Comment:
The strategy 'we keep as much of the public field as possible because we don't know which ones are used' from the start is no longer true x) But it is better in the sense that it cleans the fields that should have been protected

Reply:
Yep I changed my mind a little bit But I didn't realize there were so many public properties that are actually purely related to internal behaviour of the admin controller and that should have been protected probably